README.md
# wap4
Bản quyền By queen


news
{% use '_includes_forum' %}
{% import '_functions' as func %}
{% import '_avatar' as avatar %}
{% set url = get_uri_segments() %}
{% set signin = func.signin()|trim %}
{% set user = get_data('user_'~signin)[0].data|json_decode %}
{% set run = get_data('forum')[0].data|json_decode %}
{% set title = 'Lưu trữ tin tức' %}
{% set news = run['news']|default(0) %}
{% set smile = user['smile']|trim|split('.')|length-1 %}
{{block('head')}}
<div class="phdr"><i class="fa fa-globe" aria-hidden="true"></i> {% if url[1]=='admin.php' and user['right'] >= '8' %}{% if get_get('edit')!='' and get_get('edit') in func.get('news')|trim|split('@')|reverse %}Chỉnh sửa tin tức{% else %}Thêm tin tức{% endif %}{% else %}Lưu trữ tin tức{% endif %}</div>
{% if user['right'] >= '8' and not url[1] %}<div class="rmenu"><a href="/news/admin.php"><i class="fa fa-plus-circle" aria-hidden="true"></i> Thêm tin tức mới</a></div>{% endif %}
{% set data= func.get('news')|trim|split('@')|reverse %}
{% set total=data|length-1 %} 
 {% set page_max=total//10 %}
{% if total//10 != total/10 %}
{% set page_max=total//10+1 %}
{% endif %}
{% set p=get_get('p')|default(1) %} 
{% if p matches '/[a-zA-z]|%/' or p<1 %}
{% set p=1 %}
{% endif %}
{% if p>page_max %}
{% set p=page_max %}
{% endif %}
{% set st=p*10-10 %}
{% set index %}{% for i in data|slice(0,total)|slice(st,10) %}{% set view = get_data('news_id_'~i|trim)[0].data|json_decode %}<div class="list-login">
<big>{{view['title']}}</big><br/><small><font color="gray">Bởi: {{avatar.mau_nick(view['author'],func.get('user_'~view['author'],'right'))}}, lúc: {{func.ago(view['time'])}}</font>{% if user['right'] >= '8' %}, <a href="/news/admin.php?edit={{i|trim}}"><i class="fa fa-pencil-square-o" aria-hidden="true"></i> Chỉnh sửa</a>{% endif %}</small><div class="sub"></div>{{func.bbcode(view['doc'])}}</div>{% endfor %}{% endset %}
{% set form %}{% set token = random(100000) %}
{% set post = get_data('news_id_'~get_get('edit'))|last.data|json_decode %}
<div class="room"><div class="menu-room"><table width="100%" border="0"><tr><td width="80%"> <form action="" method="POST" name="form"><div><b>Tiêu đề:</b><input name="title" value="{% if get_get('edit')!='' and get_get('edit') in data %}{{post['title']}}{% endif %}"></div><div align="left"><b>Nội dung:</b></div>{% include 'toolbar' %}</div><textarea type= "text" name="msg" rows="8">{% if get_get('edit')!='' and get_get('edit') in data %}{{post['doc']}}{% endif %}</textarea>{% if get_get('edit')=='' %}<div align="center">{{captcha()|raw}}</div>{% endif %}<a class="sub3" href="/news/admin.php{% if get_get('edit')!='' and get_get('edit') in data %}?edit={{get_get('edit')}}{% endif %}"><i class="fa fa-refresh" aria-hidden="true"></i></a><button name="submit" type="submit">{% if get_get('edit')!='' and get_get('edit') in data %}Chỉnh sửa{% else %}Đăng tin tức{% endif %}</button><input type="hidden" name="token" value="{{token}}"/></form></div></td></tr></table></div></div>{% endset %}
{% if url[1]=='admin.php' and user['right'] >= '8' %}
{% set id_news = news|trim+1 %}
{% set title = get_post('title') %}
{% set doc = get_post('msg') %}
{% if get_get('edit') in data and get_get('edit')!='' %}
{% if request_method()|lower == "post" %} 
{% set post = get_data('news_id_'~get_get('edit'))|last.data|json_decode %}
{% if title|length >= 3 and title|length <= 100 and doc|length >= 50 and doc|length <= 16000 %}
{% set key = 'news_id_'~get_get('edit') %}
{% if post['title']!=title %}
{{func.add(key,'title',title)}}
{% endif %}
{% if post['doc']!=doc %}
{{func.add(key,'doc',doc)}}
{% endif %}
<div class="rmenu"><font color="green">Chỉnh sửa thành công!</font></div>
{% endif %}
{% endif %}
{{form}}
{% else %}
{% set data = {"title":title,"doc":doc,"author":signin,"time":"now"|date("U")} %}
{% set status = save_data_captcha('news_id_'~id_news,data|json_encode) %}
{% if request_method()|lower == "post" %} 
{% if status != 'CAPTCHA_ERR' and title|length >= 3 and title|length <= 100 and doc|length >= 50 and doc|length <= 16000 %}
{{form}}
{{func.trom_up('news',id_news,'up')}}
{{index}}
{% else %}
<div class="rmenu">Đăng tin tức thất bại! Tại sao tôi lại bị như vậy?<br/>- Có thể bạn chưa xác minh Tôi không phải là người máy<br/>- Hoặc bạn nhập nội dung không đúng với giới hạn kí tự được cho phép, gồm: <b>Tiêu đề</b> (Min.3 - Max.100), <b>nội dung</b> (Min.5 - Max.16000)</div>
{{form}}
{% endif %}
{% else %}
{{form}}
{% endif %}
{% endif %}
{% else %}
{% if total >= '1' %}
{{index|default(0)}}
{{func.paging(url|join('/')~'?p',p,page_max)}}
{% else %}
<div class="rmenu">Chưa có tin tức nào!</div>
{% endif %}
{% endif %}
{{block('end')}}

xc.php
{% import '_functions' as func %}
{% from '_functions' import get,ago,bbcode %}
{% import '_avatar' as avatar %}
{% import 'chat_module_bot' as bot %}
{% set login = func.signin()|trim %}
{% set name = login %}
{% set ubot = get_data('user_apple')[0].data|json_decode %}
{% set id = func.get('xep_chu')|split('@')[19]|trim %}
{% set msg = get_post('msg') %}
{% set now = "now"|date("U") %}
{# {% if request_method()|lower=='POST' %} #}
{% if msg != '' and msg != '\r\n' and msg!=null and msg|length <= '16000' %}
{# {{func.add('user_'~login,'new',msg)}}
{% if msg and func.get('user_'~login,'old')|trim|raw != func.get('user_'~login,'new')|trim|raw %} #}
{% set comment = {"name" :name,"time":now,"comment":msg} %}
{% if 'xep chu @apple' not in msg|lower and 'xếp chữ @apple' not in msg|lower %}
{{func.add('xc_'~id,'name',name)}}
{{func.add('xc_'~id,'time',now)}}
{{func.add('xc_'~id,'comment',msg)}} 
{{func.up('xep_chu',id,'up') }}
{% endif %}
{{func.add('user_'~login,'old',msg)}} 
{{func.add('user_'~login,'postguest',get('user_'~login,'postguest')|trim+1)}}
{{bot.xepchu(msg,'apple')}}
{# {% endif %} #}
{% endif %}
{# {% endif %} #}
{% set data= func.get('xep_chu')|trim|split('@') %}
{% set total=data|length-1 %} 
 {% set page_max=total//10 %}
{% if total//10 != total/10 %}
{% set page_max=total//10+1 %}
{% endif %}
 {% set url=get_uri_segments() %}
{% set p=url[1]|default(1) %} 

{% if p matches '/[a-zA-z]|%/' or p<1 %}
{% set p=1 %}
{% endif %}
{% if p>page_max %}
{% set p=page_max %}
{% endif %}
{% set st=p*10-10 %}
{% if ubot['xc'].time >= "now"|date("U") %}
<div class="rmenu" align="center">
{% if ubot['xc'].time=="now"|date("U") %}
Hết giờ !!!
{% else %}
Sắp xếp các chữ sau thành từ hoặc cụm từ hoàn chỉnh:<br/>{{ubot['xc'].quest}}
<br/>{% for k in (ubot['xc'].raw_word)|split('') %}{% if k != ' ' %}<font style="font-size:20px;letter-spacing:6px;">{% if (ubot['xc'].time)-("now"|date("U")) <= (loop.index-2)*(100/(ubot['xc'].raw_word|length)) %}{{k}}{% else %}*{% endif %}</font>{% else %}{{k}}{% endif %}{% endfor %}
<br/>Còn: {{(ubot['xc'].time)-("now"|date("U"))}} giây nữa
{% endif %}
</div>
{% endif %}

{% for id in data|slice(0,total)|slice(st,10) %}
{% set entry = get_data('xc_'~id|trim)[0].data|json_decode %}
{% set user='user_'~entry.name %}
{% set info=get_data(user)[0].data|json_decode %}
{% set nd = entry.comment %}
{% set time = entry.time %}
{% set jun = now-time %}
{% if jun > 1 %}
{% if time|date('d','Asia/Ho_Chi_Minh') == 'now'|date('d','Asia/Ho_Chi_Minh') %}
{% set agos = ago(time) %}
{% else %}
{% set agos = time|date("H:i, d/m/Y","Asia/Ho_Chi_Minh")|replace({(now|date("d/m/Y","Asia/Ho_Chi_Minh")):'Hôm nay'}) %}
{% endif %}
{% else %}
{% set agos = 'vừa xong' %}
{% endif %}
{% if entry.name %}
<div class="menu"><table id="chat" cellpadding="0" cellspacing="1"><tr><td width="auto"><img style="background-size: cover; background-color: #fff0; border-radius: 30px;width:30px;height:30px" src="{{avatar.avtdefault(entry.name)|trim}}" width="40" height="40" /></td><td> <a href="/user/{{entry.name}}.html">
{% if func.get(user,'ban') =='1' %}<s>{{get(user,'name')}}</s>{% else %}{{avatar.mau_nick(entry.name,info.right)}}{% endif %}
</a> <br/><i class="fa fa-clock-o"></i> {{agos}}</td></tr></table><div class="chatbox">{% if info.ban!=null %}Nội dung đã bị ẩn do người này đã vi phạm quy định của weblog{% else %}{{bbcode(nd|raw)}}{% endif %} </div> </div>
{% endif %} 
{% endfor %}
{% if ubot['xc'].time < "now"|date("U") and ubot['xc'].end != 'yes' %}
{% set id = func.get('xep_chu')|split('@')[19]|trim %}
{{func.add('xc_'~id,'name','apple')}}
{{func.add('xc_'~id,'time',ubot['xc'].time)}}
{{func.add('xc_'~id,'comment','Rất tiếc, không có ai trả lời đúng câu hỏi vừa rồi. Đáp án đúng là: [b]'~ubot['xc'].word~'[/b]')}} 
{{func.add('user_apple','xc',{"id":ubot['xc'].id|trim,"time":"now"|date("U"),"word":xc.word,"raw_word":xc.raw_word,"end":"yes"})}}
{{func.up('xep_chu',id,'up') }}
{% endif %}
{% if login and login not in func.get('show_online')|split('@') %}
{{func.up('show_online',login,'up')}}
{{func.add('user_'~login,'on','now'|date('U'))}}
{% endif %}


chat.php
{% import '_functions' as func %}
{% from '_functions' import get,ago,bbcode %}
{% import '_avatar' as avatar %}
{% import 'chat_module_bot' as bot %}
{% set login = func.signin()|trim %}
{% set name = login %}
{% set ubot = get_data('user_apple')[0].data|json_decode %}
{% set id = func.get('guestbook')|split('@')[19]|trim %}
{% set msg = get_post('msg') %}
{% set now = "now"|date("U") %}
{% if msg != '' and msg != '\r\n' and msg!=null and msg|length <= '16000' %}
{{func.add('user_'~login,'new',msg)}} 
{% if msg and func.get('user_'~login,'old')|trim|raw != func.get('user_'~login,'new')|trim|raw %}
{% set comment = {"name" :name,"time":now,"comment":msg} %}
{{func.add('chat_'~id,'name',name)}}
{{func.add('chat_'~id,'time',now)}}
{{func.add('chat_'~id,'comment',msg)}} 
{{func.up('guestbook',id,'up') }}
{{func.add('user_'~login,'old',msg)}} 
{{func.add('user_'~login,'xu',get('user_'~login,'xu')|trim+2)}}
{{func.add('user_'~login,'postguest',get('user_'~login,'postguest')|trim+1)}}
{{bot.thayphan(msg,login)}}
{{bot.chemgio(msg,login)}}
{{bot.minigame(msg,'apple')}}
{{bot.ott(msg,'apple')}}
{% endif %}
{% endif %}
{% set data= func.get('guestbook')|trim|split('@') %}
{% set total=data|length-1 %} 
 {% set page_max=total//10 %}
{% if total//10 != total/10 %}
{% set page_max=total//10+1 %}
{% endif %}
 {% set url=get_uri_segments() %}
{% set p=url[1]|default(1) %} 

{% if p matches '/[a-zA-z]|%/' or p<1 %}
{% set p=1 %}
{% endif %}
{% if p>page_max %}
{% set p=page_max %}
{% endif %}
{% set st=p*10-10 %}
{% for id in data|slice(0,total)|slice(st,10) %}
{% set entry = get_data('chat_'~id|trim)[0].data|json_decode %}
{% set user='user_'~entry.name %}
{% set info=get_data(user)[0].data|json_decode %}
{% set nd = entry.comment %}
{% set time = entry.time %}
{% set jun = now-time %}
{% if jun > 1 %}
{% if time|date('d','Asia/Ho_Chi_Minh') == 'now'|date('d','Asia/Ho_Chi_Minh') %}
{% set agos = ago(time) %}
{% else %}
{% set agos = time|date("H:i, d/m/Y","Asia/Ho_Chi_Minh")|replace({(now|date("d/m/Y","Asia/Ho_Chi_Minh")):'Hôm nay'}) %}
{% endif %}
{% else %}
{% set agos = 'vừa xong' %}
{% endif %}
{% if entry.name %}


<div class="menu"><table id="chat" cellpadding="0" cellspacing="1"><tr><td width="auto"><img class="avt" src="{{avatar.avtdefault(entry.name)|trim}}" width="40" height="40" /></td><td><span name="online">{% if info['on'] < ('now'|date('U')-300) %}<font color="red"><i class="fa fa-toggle-off" aria-hidden="true"></i></font>{% else %}<font color="green"><i class="fa fa-toggle-on" aria-hidden="true"></i></font>{% endif %}</span> <a href="/user/{{entry.name}}.html">
{% if func.get(user,'ban') =='1' %}<s>{{get(user,'name')}}</s>{% else %}{{avatar.mau_nick(entry.name,info.right)}}{% endif %}
</a> <br/><i class="fa fa-clock-o"></i> {{agos}} {% if login!=entry.name and login %}<br/><a href="javascript:tag('@{{entry.name}} ', '')">[Tag]</a> <a href="javascript:tag('[quote]{{id|trim}}[/quote] ', '')">[Trả lời]</a>{% endif %}</td></tr></table><div class="chatbox"><span class="textchat"> </span> {% if info.ban!=null %}Nội dung đã bị ẩn do người này đã vi phạm quy định của weblog{% else %}{{bbcode(nd|raw)}}{% endif %} </div> </div>

{% endif %} 
{% endfor %}
{% if login and login not in func.get('show_online')|split('@') %}
{{func.up('show_online',login,'up')}}
{{func.add('user_'~login,'on','now'|date('U'))}}
{% endif %}


login.php
{% use '_includes_forum' %}
{% import '_functions' as func %}
{% set url = get_uri_segments() %}
{% set signin = func.signin()|trim %}
{% set run = get_data('forum')[0].data|json_decode %}
{% set title = 'Đăng nhập | Cà Phê Đắng' %}
{% if signin %}
<script language="javascript" type="text/javascript"> 
window.location.href="/community"; 
</script> 
<META HTTP-EQUIV="Refresh" CONTENT="0;URL=/community">
{% else %}
{{block('head')}}
{% set user=get_post('user') %}
{% set pass=get_post('pass') %}
<div class="phdr">Đăng nhập</div>
<div class="menu">
{% if request_method()|lower == "post" %} 
{% if user and pass %}
{% if get_data_count('user_'~func.rwurl(user))==0 %}
<p><font color="red">Tài khoản không tồn tại.</font></p>
{% else %}
{% if func.get('user_'~func.rwurl(user),'pass')!=func.ma_hoa(pass)|trim %}
<p><font color="red">Mật khẩu không đúng.</font></p>
{% else %}
<p><font color="green">Đăng nhập thành công.</fon></p>
{{ set_cookie('token',func.get('user_'~func.rwurl(user),'token')|trim) }} 
<script language="javascript" type="text/javascript"> 
window.location.href="/community"; 
</script> 
<META HTTP-EQUIV="Refresh" CONTENT="0;URL=/community">
{% endif %}
{% endif %}
{% else %}
<p><font color="red">Vui lòng điền đầy đủ thông tin!</font></p>
   {% endif %}
{% endif %}
<form method="post" action="">
<p>Tài khoản:</p>
<input class="form-control" type="text" name="user" placeholder="Nhập tài khoản" autofocus>
<p>Mật khẩu:</p>
<input class="form-control" type="password"  name="pass" placeholder="Nhập mật khẩu" autofocus>
<p><b style="color:red">*</b> Sử dụng tài khoản CàFêĐắng để đăng nhập !</p>
<p><input class="btn btn-primary" type="submit" name="submit" value="Đăng nhập"></p>
</form>
</div>
{#{% set checkpass = func.get('bot','pass') %}#}
{{block('end')}}
{% endif %}

logout.php
{% use '_includes_forum' %}
{% import '_functions' as func %}
{% set signin = func.signin()|trim %}
{% set title = 'Đăng xuất | Cà Phê Đắng' %}
<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta http-equiv="content-language" content="vi" />
<script type="text/javascript">if(navigator.userAgent.match(/iPhone/i)||navigator.userAgent.match(/iPad/i)){var viewportmeta=document.querySelector('meta[name="viewport"]');viewportmeta&&(viewportmeta.content="width=device-width, minimum-scale=1.0, maximum-scale=1.0, initial-scale=1.0",document.body.addEventListener("gesturestart",function(){viewportmeta.content="width=device-width, minimum-scale=0.25, maximum-scale=1.6"},!1))}; </script>
<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1.0,user-scalable=yes">
<meta name="HandheldFriendly" content="true" />
<meta name="MobileOptimized" content="width" />
<title>{{title}}</title>
</head>
<body>
{% if signin %}
{{delete_cookie('token')}}
Đăng xuất tài khoản thành công!
<script language="javascript" type="text/javascript"> 
window.location.href="/community"; 
</script> 
<META HTTP-EQUIV="Refresh" CONTENT="0;URL=/community">
{% endif %}
</body>
</html>


search.php
{% use '_includes_forum' %}
{% from 'func.twig' import mi_get,mi_up,k_del,get,html_decode %}
{% from 'function.twig' import add,mi_add,slug,time,description %}
{% import '_functions' as func %}
{% set url = get_uri_segments() %}
{% set data = mi_get('show_blog')|split(' @ ')|reverse  %}
{% set domain = current_url|split('/').0~"//"~current_url|split('/').2 %}
{% set data1 %}{% for i in data %}{% set key = get_get('key') %}
{% if key|length>'0' and key|lower in get('blog_'~i|trim,'title')|trim|lower or key|length>'0' and key|lower in get('blog_'~i|trim,'content')|trim|lower %}{% if loop.last==false %}{{i|trim}}.{% endif %}{% endif %}{% endfor %}{% endset %}
{% set count = data1|split('.')|length-1 %}
{% if get_get('key')!='' %}
{% set title = 'Kết quả tìm kiếm "'~get_get('key')~'" ' %}
{% else %}
{% set title = 'Tìm kiếm' %}
{% endif %}
{% set data=data1|split('.') %}
 {% set total=data|length-1 %} 
{% set per = '10' %}
 {% set page_max=total//per %}
{% if total//per != total/per %}
{% set page_max=total//per+1 %}
{% endif %}
{% set p=get_get('p')|default('1') %} 
{% if p matches '/[a-zA-z]|%/' or p<1 %}
{% set p=1 %}
{% endif %}
{% if p>page_max %}
{% set p=page_max %}
{% endif %}
{% set st=p*per-per %}
{% set description %}{% for id in data|slice(0,total)|slice(st,per) %}
{% set blog = 'blog_'~id|trim %}{% set title2 = get(blog,'title') %}{% set pre = title2|split('[')[1]|split(']')[0] %}{% if pre!='' %}{{title2|replace({(pre):'','[':'',']':''})}}{% else %}{{title2}}{% endif %}, {% endfor %}{% endset %}
{% set description = description|slice(0,250)|trim %}
{{block('head')}}
<div class="phdr border_red">Tìm kiếm</div>
<div class="menu" style="text-align: center;"><form action="" method="get"><input type="text" id="search-input" name="key" value="{{get_get('key')}}" placeholder="Nhập từ khóa cần tìm..."><button type="submit"><i class="fa fa-search" aria-hidden="true"></i> Tìm</button></form></div>
<div id="results-container"></div>
{% if count > '0' %}
<div class="newsx" align="center"><i class="fa fa-thumbs-o-up" aria-hidden="true"></i> Có {{count}} kết quả được tìm thấy</div>
{% else %}
{% if get_get('key')|length > 0 %}
<div class="newsx" align="center"><i class="fa fa-thumbs-o-down" aria-hidden="true"></i> Không tìm thấy kết quả nào </div>
{% endif %}
{% endif %} 
{% for id in data|slice(0,total)|slice(st,per) %}
{% set blog = 'blog_'~id|trim %}
{% set title = get(blog,'title') %}
{% set slug = get(blog,'slug') %}
<div class="list1 pre"><i class="fa fa-rss" aria-hidden="true"></i> <a href="/library/{{id|trim}}-{{slug}}">{{func.pre(title)}}</a></div>
{% endfor %}
{% if page_max>per %}
{% set page_max=per %} 
 {% endif %} 
 {{func.paging(url|join('/')~'?key='~get_get('key')~'&p',p,page_max)}}
{{block('end')}}


registration.php
{% use '_includes_forum' %}
{% import '_functions' as func %}
{% from 'function.twig' import mi_add %}
{% from 'func.twig' import get,mi_get,html_decode %}
{% set url = get_uri_segments() %}
{% set signin = func.signin()|trim %}
{% set run = get_data('forum')[0].data|json_decode %}
{% if get_get('act')=='faq' %}
{% set title = 'Quy định sử dụng' %}
{% else %}
{% set title = 'Đăng ký tài khoản | Cà Phê Đắng' %}
{% endif %}
{% if signin %}
<script language="javascript" type="text/javascript"> 
window.location.href="/community"; 
</script> 
<META HTTP-EQUIV="Refresh" CONTENT="0;URL=/community">
{% else %}
{{block('head')}}
{% if get_get('act')=='faq' %}
<div class="phdr"><b>Giới thiệu</b></div>
<div class="menuItem" style="background-color:#e9ccd2;">
<b>CàFêĐắng</b> là web chat giải trí dành cho các thành viên, tại đây bạn thoải mái giao lưu, kết bạn, nhắn tin và đăng chủ đề, mua bán và nhiều tiện ích khác!
</div>
<div class="phdr"><b>Quy Định Sử Dụng</b></div>
<div class="list-login"><p><b>1) CẤM:</b><br/><br/><b>1.1</b> Đăng tin quảng cáo.<br/><b>1.2</b> Đăng nội dung liên quan đến vấn đề phân biệt sắc tộc, tôn giáo, quốc gia.<br/><b>1.3</b> Không đăng nội dung thô tục, khiêu dâm, quấy rối tình dục, vu khống, đe dọa người khác.<br/><b>1.4</b> Không được đăng nội dung xúc phạm một chủ thể cụ thể, đặc biệt là thành viên trong diễn đàn.<br/><b>1.5</b> Cố tình giả mạo thành viên khác với bất kỳ mục đích nào.<br/><b>1.6</b> Đăng bài cùng một nội dung nhiều lần trong một hay nhiều chủ đề khác nhau.<br/><b>1.7</b> Đăng lại bài viết hoặc tạo lại chủ đề đã bị xóa bởi Ban quản trị.<br/><b>1.8</b> Đăng bài có nội dung không liên quan đến tiêu đề.<br/><br/><b>2) TUYỆT ĐỐI CẤM:</b><br/><br/><b>2.1</b> Tạo chủ đề than phiền về thành viên hoặc cách hoạt động của điều hành viên. Hãy sử dụng tin nhắn riêng cho vấn đề này<br/><b>2.2</b> Đăng ký nick chứa ký tự gây khó chịu cho người khác<br/><br/><b>3) KHÔNG CHẤP NHẬN:</b><br/><br/><b>3.1</b> Trích dẫn nội dung quá dài hoặc trích dẫn lại một bài đăng chứa một trích dẫn khác.<br/><b>3.2</b> Bài viết tin chỉ chứa liên kết đến các trang web khác. <br/><b>3.3</b> Đăng bài bằng ngôn ngữ không sử dụng chữ cái latin (a-z).<br/><b>3.4</b> Đăng bài không có mục đích: chỉ có biểu tượng vui hoặc bài như "Ok", "Có ai không?",... <br/><br/>Quy định trên đây có thể thay đổi bất kỳ lúc nào.<br />Quy định này áp dụng cho tất cả các thành viên và Ban quản trị, thành viên vi phạm nhiều lần sẽ bị ban, vi phạm nặng có thể dẫn đến xóa bỏ tài khoản vĩnh viễn.</p></div>
<div class="menuItem" align="center">
Click <a href="/{{url[0]}}">đăng ký</a> nếu bạn đã đọc, hiểu rõ và sẽ chấp hành những quy định trên của cộng đồng !</div>
{% else %}
<div class="phdr">Đăng ký</div>
{% if run.account <= '500' %}
<div class="gmenu">
{# kiểm tra và lưu tài khoản #}
{% set user = get_post('user') %}
{% set pass = get_post('pass') %} 
{% set repass = get_post('repass') %} 
{% set mbv = get_post('captcha') %}
{% set captcha = get_cookie('captcha')|lower %}
{% set token=func.token()|trim %}
{% set registration %}
<p>{{block('mem')}}</p>
<form method="post" action="">
<p>Tài khoản:</p>
<input class="form-control" type="text" name="user" value="" required>
<p><div class="rmenu"><b style="color:red">*</b> Tên tài khoản khi đăng ký không vượt quá 15 kí tự và tối tiểu phải đặt 3 kí tự, mật khẩu dài tùy ý.</div></p>
<p>Mật khẩu:</p>
<input class="form-control" type="password"  name="pass" value="" required>
<p>Nhập lại mật khẩu:</p>
<input class="form-control" type="password" name="repass" value="" required>
<p>Mã xác nhận: [ <b>{{captcha}}</b> ]</p>
<input class="form-control" type="number" name="captcha" value="" required>
<p>{{captcha()|raw}}</p>
<p><input class="btn btn-primary" type="submit" name="submit" value="Đăng ký"></p></form>
{% endset %}
{% if request_method()|lower == "post" %}
{% if user and pass and repass and mbv %} 
{% if ("now"|date("U") < run.time_reg )%}   
<p><font color="red">Xin lỗi vì sự cố này, bạn có thể tiếp tục đăng ký tài khoản sau {{run.time_reg - "now"|date("U")}} giây</font></p>
{% elseif pass!=repass %}
<p><font color="red">Mật khẩu xác nhận không đúng.</font></p>
{% elseif captcha!=mbv %}
<p><font color="red">Mã xác nhận không đúng.</font></p>
{% else %}
{% if get_data_count('user_'~func.rwurl(user))>0 %} 
<p><font color="red">Tài khoản đã tồn tại.</font></p>
{{registration}}
{% elseif user|length<3 or user|length>15 %}
<p><font color="red">Tài khoản không dài quá 15 kí tự, tối thiểu 3 kí tự</font></p>
{{registration}}
{% elseif user=='admin' %}
<p><font color="red">Bạn không thể đăng ký sử dụng tài khoản này !</font></p>
{{registration}}
{% else %} 
{% if user matches '/^[a-zA-Z0-9\\-\\_]+[a-zA-Z0-9\\-\\_]$/' %} 
{# xác minh người máy #}
{% set data={"id":run.account|trim+1,"name":user,"nick":user,"pass":func.ma_hoa(pass)|trim,"right":"0","xu":"5000","luong":"10","avatar":"0.0.0.0.0.0.0.0.0.0.0.0.0","avt":"x1","abia":1,"token":token,"reg":"now"|date("U"),"cb":"autoload","chest":{"ao":"0.","quan":"0.","canh":"0.","docamtay":"0.","khac":"0.","khuonmat":"0.","kinh":"0.","mat":"0.","matna":"0.","non":"0.","nen":"0.","thucung":"0.","toc":"0."},"hp":{"min":"50","max":"50"},"sm":"100"} %}
{% set status = save_data_captcha("user_"~func.rwurl(user),data|json_encode) %}
{% if status == 'CAPTCHA_ERR' %}
<p><font color="red">Vui lòng xác minh Tôi không phải là người máy!</font></p>
{{registration}}
{% else %}
<div class="f">
<p><b><big>Dữ liệu thông tin:</big></b></p>
<p><b>Tài khoản:</b> {{user}} </p>
<p><b>Mật khẩu:</b> {{pass}} </p>
<p><a href="/login.php"><button class="btn btn-primary">Đăng nhập</button></a> | <a href="/community">Cộng đồng</a></p>
</div>
{% set old_token = html_decode(get('token'))|replace({'”':'"'}) %}
{% set new_token = old_token|json_decode|merge({(token):(user)}) %}
{{mi_add('token',(new_token))}}
{#{{func.add('token',token,user)}}#}
{{func.add('forum','new_mem',user)}}
{{func.add('forum','time_reg',"now"|date("U")+50)}}
{{func.add('forum','account',run.account|trim+1)}}
{{func.up('member',func.rwurl(user),'up')}}
{{set_cookie('token',token)}}
{{set_cookie('captcha',random(1000..9999))|lower}}
{% endif %}
{# kết thúc xác minh #}
{% else %}
<p><font color="red">Tài khoản không được chứa ký tự đặc biệt.</font></p>
{{registration}}
{% endif %} 



{% endif %}
{% endif %}
{% else %}
<p><font color="red">Vui lòng điền đầy đủ thông tin và xác minh Tôi không phải là người máy!</font></p>
{{registration}}
{% endif %}
{% else %}
{{registration}}
{% endif %}
</div>
{% else %}
<div class="rmenu menuItem">
Đăng ký tạm thời đóng cửa !
</div>
{% endif %}
{% endif %}
{{block('end')}}
{% endif %}